bpfd/src/serve.rs

```diff
@@ -150,7 +145,12 @@ async fn serve_unix(
 
     let uds = UnixListener::bind(&path)?;
     let uds_stream = UnixListenerStream::new(uds);
-    set_file_permissions(&path, SOCK_MODE).await;
+    // Always set the file permissions of our listening socket.
+    if (tokio::fs::set_permissions(path.clone(), std::fs::Permissions::from_mode(SOCK_MODE)).await)
```

Comment:
The function only sets permissions on the socket if it's in the bpfd user group https://github.com/bpfd-dev/bpfd/blob/main/bpfd/src/utils.rs#L63

We should ALWAYS be setting the socket perms correctly since it's used by processes other than bpfd.